internal/signing/signature_client.go

```diff
@@ -0,0 +1,63 @@
+package signing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/network"
+	"github.com/stellar/go/txnbuild"
+)
+
+var ErrInvalidTransaction = errors.New("invalid transaction provided")
+
+type SignatureClient interface {
+	NetworkPassphrase() string
+	GetDistributionAccountPublicKey() string
+	SignStellarTransaction(ctx context.Context, tx *txnbuild.Transaction) (*txnbuild.Transaction, error)
+}
+
+type envSignatureClient struct {
+	networkPassphrase       string
+	distributionAccountFull *keypair.Full
+}
```

Comment:
I recommend you to override the String() method, in order to prevent distributionAccountFull from ever being printed.

It should be overridden in the value and not the pointer, like in:

https://github.com/stellar/stellar-disbursement-platform-backend/blob/d3912e908b6c1cea56e7aad4b79b18b7e2c583e4/internal/transactionsubmission/store/db_vault.go#L22-L24



Here are the tests that prove it should be implemented in the value. They'd not pass if the implementation was in the pointer:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/d3912e908b6c1cea56e7aad4b79b18b7e2c583e4/internal/transactionsubmission/store/db_vault_test.go#L32-L64